kratos/CMakeLists.txt

```diff
@@ -100,6 +101,7 @@ set( KRATOS_CORE_SOURCES
     ${CMAKE_CURRENT_SOURCE_DIR}/conditions/mesh_condition.cpp;
     ${CMAKE_CURRENT_SOURCE_DIR}/processes/apply_periodic_boundary_condition_process.cpp;
     ${CMAKE_CURRENT_SOURCE_DIR}/linear_solvers/amgcl_solver_impl.cpp;
+    ${CMAKE_CURRENT_SOURCE_DIR}/utilities/mpi_coloring_utilities.cpp
```

Comment:
ouch...

now we have an MPI-core, why not putting it there?
If I understand correctly this is only needed for MPI right?

Reply:
the mpi_coloring_utilities.cpp does not have any dependency on mpi (it only depends on DataCommunicator) and is included in the pointer_communicator which can be used both in serial and in parallel code (so when the mpi core is not loaded)

this means that as a utility it needs to go in the core and in the mpi_core

Reply:
I did not understand what do you mean by 
> this means that as a utility it needs to go in the core and in the mpi_core

Do we need to add it to both libraries?

Also, if it has no relation to mpi, why is it called `mpi_....`?

Reply:
the phrase should have been:
in the core and NOT in the mpi_core.

just think: pointer_comm can be used when you have no mpi right? it includes the mpi_coloring_utilities.h , so this file needs to be available when you compile the pointer_comm.

maybe we could rename it as 

mpi_coloring_utilities -> communication_coloring_utilities

Reply:
:+1: for the rename.

Reply:
@jcotela i would prefer to make the renaming in a different PR as soon as this is merged. 

the point is that here i already moved the mpi_coloring_utilities, and i would really like to avoid further messing up the same file.

Reply:
(note that i am referring to renaming the name of the file)